src/snowflake/cli/plugins/object/manager.py

```diff
@@ -52,3 +55,83 @@ def object_exists(self, *, object_type: str, name: str):
             return True
         except ProgrammingError:
             return False
+
+    def _send_rest_request(
+        self, url: str, method: str, data: Optional[Dict[str, Any]] = None
+    ):
+        # SnowflakeRestful.request assumes that API response is always a dict,
+        # which is not true in case of this API, so we need to do this workaround:
+        from snowflake.connector.network import (
+            CONTENT_TYPE_APPLICATION_JSON,
+            HTTP_HEADER_ACCEPT,
+            HTTP_HEADER_CONTENT_TYPE,
+            HTTP_HEADER_USER_AGENT,
+            PYTHON_CONNECTOR_USER_AGENT,
+        )
+
+        self._log.debug(f"Sending {method} request to {url}")
+        rest = self._conn.rest
+        full_url = f"{rest.server_url}{url}"
+        headers = {
+            HTTP_HEADER_CONTENT_TYPE: CONTENT_TYPE_APPLICATION_JSON,
+            HTTP_HEADER_ACCEPT: CONTENT_TYPE_APPLICATION_JSON,
+            HTTP_HEADER_USER_AGENT: PYTHON_CONNECTOR_USER_AGENT,
+        }
+        return rest.fetch(
+            method=method,
+            full_url=full_url,
+            headers=headers,
+            token=rest.token,
+            data=json.dumps(data if data else {}),
+            no_retry=True,
+        )
+
+    def _url_exists(self, url):
+        try:
+            result = self._send_rest_request(url, method="get")
+            return bool(result) or result == []
+        except InterfaceError as err:
+            if "404 Not Found" in str(err):
+                return False
+            raise err
+
+    def _get_rest_api_create_url(self, object_type: str):
+        plural_object_type = _get_object_names(object_type).sf_plural_name.replace(
```

Comment:
This is based on `OBJECT_TO_NAMES` which is a hardcoded list of supported objects. It does not sounds like something we agreed on (having hardcoded list).

Reply:
I used that as it is a "sure way" of correctly pluralize the type, but I see what you mean.

Looking at a list of currently supported objects, I replaced that with a simple heuristic